File: scribbleshare-app/scribbleshare-frontend/src/scripts/canvas/canvasObject/EntityCanvasObject.js
import CanvasObject from "./CanvasObject.js";

export default class EntityCanvasObject extends CanvasObject {
    constructor(reader) {
        super(reader);
        this.width = reader.readInt16();
        this.height = reader.readInt16();
        this.rotation = reader.readUint8();
    }

    serialize(writer) {
        super.serialize(writer);
        writer.writeInt16(this.width);
        writer.writeInt16(this.height);
        writer.writeUint8(this.rotation);
    }
}

File: scribbleshare-commons/src/main/java/net/stzups/scribbleshare/data/objects/canvas/canvasObject/EntityCanvasObject.java
package net.stzups.scribbleshare.data.objects.canvas.canvasObject;

import io.netty.buffer.ByteBuf;
import net.stzups.scribbleshare.util.DebugString;

public class EntityCanvasObject extends CanvasObject {
    private final short width;
    private final short height;
    private final byte rotation;

    public EntityCanvasObject(ByteBuf byteBuf) {
        super(byteBuf);
        width = byteBuf.readShort();
        height = byteBuf.readShort();
        rotation = byteBuf.readByte();
    }

    @Override
    public void serialize(ByteBuf byteBuf) {
        super.serialize(byteBuf);
        byteBuf.writeShort(width);
        byteBuf.writeShort(height);
        byteBuf.writeByte(rotation);
    }

    @Override
    public String toString() {
        return DebugString.get(EntityCanvasObject.class, super.toString())
                .add("width", width)
                .add("height", height)
                .add("rotation", rotation)
                .toString();
    }
}
